ragtime.jdbc/src/ragtime/jdbc.clj

```diff
@@ -17,10 +17,18 @@
                         [[:id "varchar(255)"]
                          [:created_at "varchar(32)"]]))
 
-(defn- ensure-migrations-table-exists [db-spec migrations-table]
-  (try
-    (sql/execute! db-spec [(migrations-table-ddl migrations-table)])
-    (catch SQLException _)))
+(defn- table-exists?
+  [connection table-name]
+  (when connection
+    (let [results (.getTables (.getMetaData connection) nil nil table-name nil)
```

Comment:
Unfortunately, using metadata-query won't preclude the direct use of some Java functions (specifically the `.getMetaData` and `.getTables` calls). 

If we try to use, e.g. `with-db-metadata` or `metadata-query`, any metadata retrieval is handled within the context of a connection, which is closed afterwards. This causes other tests to fail and generally isn't transaction-friendly. 

We can get around the `.next` and `.getString` calls, however, using `metadata-result`.